### ISSUE TITLE
TemplateSceptreHandlerError is too generic

### ISSUE THREAD

**Issue by mikkoc** [cloudreach/sceptre#129]
Despite having the correct handler function, I kept hitting this error:
`The template does not have the required 'sceptre_handler(sceptre_user_data)' function.`

It turns out the real error was:
`AttributeError: AWS::EC2::NatGateway object does not support attribute Tags`

but sceptre was masking it by raising a generic TemplateSceptreHandlerError.

Code that was causing the issue:
```
    def add_natgw(self):
        self.natgw = self.template.add_resource(NatGateway(
            "NatGateway",
	    AllocationId=Ref(self.eip_id),
	    SubnetId=Ref(self.subnet_id),
            Tags=[
                Tag("Name", Join("-", [Ref(self.environment), "nat-gateway"]))
            ],
        ))
```

After removing the Tags attribute, I managed to get past the error.

Sceptre should be more specific and print the underlying error.

**Comment by theseanything** [comment]
(I had sent me the full troposphere template earlier.) Issue was using spaces as indention within the statement above. This syntax error made it such that sceptre_handler function wasn't correctly defined. Hence removing the statement above fixed the issue. 

When indented correctly, a `AttributeError` is throw as it should.

**Comment by danmx** [comment]
+1
if there is an issue with a template or your template does some weird stuff that are not supported by Sceptre but run perfectly as stand-alone you only get this error message:
`The template does not have the required 'sceptre_handler(sceptre_user_data)' function.`

**Comment by jamesroutley** [comment]
@danmx Yep - we've got a fix for that #137 merged into master. We'll be releasing it shortly.
